### DISCUSSION TITLE
Customizable Brush Events?

### DISCUSSION THREAD

**Discussion by spren9er** [uwdata/mosaic#189] [General]
When dealing with lots of data, brushing on `brush` events for `interval1d` and `interval2d` interactors can lead to lagging queries and sluggish UI:

https://github.com/uwdata/mosaic/assets/5602551/45d59b5f-0afb-47a3-8409-26dd2bc4890e

In that case, switching to brushing only on `end` events would be better:

https://github.com/uwdata/mosaic/assets/5602551/45647814-3bba-40de-8ca7-154e08bc8abe

My suggestion would be to introduce a new attribute `brushEvent` to `intervalX`, `intervalY` and `intervalXY` interactors, which is by default `brush end`, but could be changed to `end`.

Any other ideas?

**Comment by jheer** [comment]
That sounds reasonable to me. One question is if we want to expose these d3-specific event names, versus having a bit more abstraction via a boolean toggle (default `false`), which when `true` results in `end` events only. Are there any cases other than `brush end` and `end` that we might want to consider?

**Reply by spren9er** [reply]
I agree, a boolean toggle would be sufficient and also be more user-friendly.
If I find some time, I'll implement this...
